%----------------------------------------------------------------------------------------
%	MODULE INFORMATION
%----------------------------------------------------------------------------------------

% Define the top matter
\setModuleTitle{Cohort analysis for the identification of driver genes}
\setModuleAuthors{%
  Ann-Marie Patch, QIMRBerghofer \mailto{ann-marie.patch@qimrberghofer.edu.au} \\
  Erdahl Teber, CMRI \mailto{eteber@cmri.org.au}%
}

\setModuleContributions{%
  Scott Wood \mailto{scott.wood@qimrberghofer.edu.au}%
}

%----------------------------------------------------------------------------------------
%	MODULE TITLE PAGE
%----------------------------------------------------------------------------------------

\chapter{\moduleTitle}

%----------------------------------------------------------------------------------------

\newpage

%----------------------------------------------------------------------------------------
%	LEARNING OUTCOMES
%----------------------------------------------------------------------------------------

\section{Key Learning Outcomes}

After completing this practical the trainee should be able to:

\begin{itemize}
  \item Run the IntOGen analysis software on cohort mutation data.
  \item Have gained experience of the structure of the analysis output files in order to identify potential driver genes.
  \item Have gained overview knowledge of different methods for identification of genes important in cancers.
\end{itemize}

%----------------------------------------------------------------------------------------
%	MODULE RESOURCES
%----------------------------------------------------------------------------------------

\section{Resources You'll be Using}

\subsection{Tools Used}

\begin{description}[style=multiline,labelindent=0cm,align=left,leftmargin=1cm]
  \item[IntOGen mutations platform] \hfill\\
    \url{https://www.intogen.org/search}
\end{description}

%------------------------------------------------

\subsection{Sources of Data}

\begin{description}[style=multiline,labelindent=0cm,align=left,leftmargin=1cm]
 \item[TCGA melanoma somatic SNV data from 338 tumour samples] \hfill\\
  \url{https://tcga-data.nci.nih.gov/tcga/}
\end{description}
%----------------------------------------------------------------------------------------

%------------------------------------------------

\subsection{Useful Links}

\begin{description}[style=multiline,labelindent=0cm,align=left,leftmargin=1cm]
 \item[Mutation Annotation Format (MAF) specification] \hfill\\
  \url{https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification}
\end{description}
%----------------------------------------------------------------------------------------

\newpage

%----------------------------------------------------------------------------------------
%	INTRODUCTION
%----------------------------------------------------------------------------------------

\section{Introduction}
Cancer driver genes are commonly described as genes that when mutated directly affect
the potential of a cell to become cancerous. They are important to a tumour cell as they
confer a growth or survival advantage over the normal surrounding cells. The mutations
in these driver genes are then clonally selected for as the population of tumour cells
increases. We think of the key genes driving tumour initiation (development), progression,
metastases, resistance and survival. Driver gene mutations are often described as "early"
events because they were key in turning a normally functioning and regulated cell into a
dysregulated one. The logical assumption is that these key mutations will be present in
all tumour cells in a patient's sample; although sometime this is not true.
\vspace{4 mm}
There are two major research goals that underline the need to identify driver genes:
\begin{itemize}
\item By identifying the early changes that take place researchers might be able
to find a treatment to stop the root cause of why cells become malignant. 
\item By identifying groups of patients with the same genes mutated then we can
develop therapies that will work for all of them.
\end{itemize}
\vspace{4 mm}
When we sequence tumour samples we tend to use samples that come from fully developed
cancers that can carry hundreds to thousands of mutations in genes and many more outside
of genes. The accumulation of these passenger mutations in cancer cells can happen
because often the repair mechanisms or damage sensing processes are amongst the first
pathways to become disrupted accelerating the mutational rate. Mutations that occur in
genes after the cell has become cancerous may still affect the growth rate, invasiveness
and even the response to chemotherapy but may not be present in all cells of a tumour.
These genes may be drivers of chemo-resistance or metastasis and are equally good targets
for therapies.
\vspace{4 mm}
IntOGen-mutations is a platform that aims to identify driver mutations using two
methodologies from cancer cohort mutation data: the first identifies mutations that
are most likely to have a functional impact combined with identifying  genes that are
frequently mutated; and the second, genes that harbour clustered mutations. These measures
are all indicators of positive selection that occurs in cancer evolution and may help
the identification of driver genes.

%----------------------------------------------------------------------------------------
%	analysing cohort data with IntOGen
%----------------------------------------------------------------------------------------

\section{Analysing cancer cohort data with IntOGen}

\begin{information}
IntOGen-mutations is a web platform that can allow users to run their analysis on the
host's servers or it can be downloaded and run without any limits on the number of
analyses on a local server.
\vspace{4 mm}
For the purposes of the course we will be using a local version of IntOGen so that we
don't encounter any issues sharing resources.
\end{information}

\begin{steps}
To start IntOGen open a terminal and navigate to the somatic/intogen\_mutations\_analysis directory.
\begin{lstlisting}
cd ~/somatic/intogen_mutations_analysis
\end{lstlisting}
\vspace{4 mm}
start IntOGen by typing
\begin{lstlisting}
./run web
\end{lstlisting}
\end{steps}

\begin{information}
A browser window should open up with the IntOGen home page displayed. 
IntoGen requires you to log in to Mozilla persona.
\vspace{4 mm}
On the top right hand side of the screen there will be a "Sign in" button.
\end{information}

\begin{steps}
Click the \texttt{Sign in} button.
\end{steps}

If you can access your email from the web you can create yourself an account. 
You do need to verify your email address by logging in to your webmail.
\vspace{4 mm}
Alternatively, you can log in using the general account details supplied in class.

\begin{steps}
Click on the \texttt{Analysis} button on the top tool bar. 
Click on the \texttt{Analyse your data} button in the \texttt{Cohort analysis} panel.
\vspace{4 mm}
A form will open up, click on the \texttt{Load} button and navigate to the
TCGA\_Melanoma\_SMgene.maf file located in the directory path 
\texttt{somatic/intogen/ TCGA\_Melanoma\_SMgene.maf}.
\vspace{4 mm}
Ensure the Genome assembly is set to \texttt{hg19(GRCh37)}.
\vspace{4 mm}
The click the \texttt{Start analysis} button.
\end{steps}

\begin{note}
The TCGA melanoma maf used in this practical has been modified from the original
to reduce processing time and only contains data for the top 680 mutated genes.
\end{note}

\begin{information}
The tool will take around 10 minutes to run and the progress will be indicated
by the green bar. You may need to refresh the browser if it looks like there
has been no progress for a while.
\vspace{4 mm}
Once complete the output can be downloaded.
\end{information}

\begin{steps}
Click on the download link to the left of the green bar or go through the
results button at the top of the page.
\vspace{4 mm}
Select to \texttt{save} the file which will end up in the download
directory \texttt{~/Downloads}.
\vspace{4 mm}
Click on the blue download arrow at the top right hand side of the browser window.
\vspace{4 mm}
Click on the file called \texttt{TCGA\_Melanoma\_slimSMgene.zip}.
\vspace{4 mm}
Highlight the file name in the pop-up window and click on \texttt{Extract}.
\vspace{4 mm}
Select the \texttt{/trainee/somatic/intogen/} directory and click on the
\texttt{Extract} button at the bottom of the window.
\vspace{4 mm}
Click to show files once extracted.
\vspace{4 mm}
Double click on the directory called \texttt{TCGA\_Melanoma\_slimSMgene}.
\end{steps}

%----------------------------------------------------------------------------------------
%	Exploring the output of IntOGen
%----------------------------------------------------------------------------------------
\section{Exploring the output of IntOGen}

\begin{information}
When you run your data over the web on the remote site there is a browse facility that
allows you to explore your data using the web version of the database. Running IntOGen locally
provides the same tabular information but in a flat file format.
\end{information}

\begin{information}
There are 7 files generated in a successful run of IntOGen:
\begin{itemize}
 \item genes.tsv - this is the main output summary table.
 \item variant\_genes.tsv - describes the variants identified in the genes
 \item variant\_samples.tsv - lists the variants together with the samples Ids
 \item pathways.tsv - lists the Kegg pathway ID for perturbed pathways
 \item consequences.tsv - lists consequences of the variants in all known transcripts
 \item project.tsv - one line to summarise the input data
 \item fimpact.gitools.tdm - the output of the functional impact tool
\end{itemize}
\end{information}

The scope of this practical will concentrate on identification of driver genes so we
will look at the main output concerning genes. OncodriveFM does however also
calculates a functional impact bias of high impact mutations in annotated Kegg
pathways \url{http://www.genome.jp/kegg/pathway.html} as shown in the
\texttt{pathways.tsv} file but will not be used in this course.

\begin{steps}
Open up the \texttt{genes.tsv} in the spreadsheet software by double clicking on the file.
\end{steps}

\begin{information}
This file contains the overall summary results for the IntOGen pipeline presented by
gene and reports Q values (i.e. multiple testing corrected P values) for the mutation
frequency and cluster modules.
\vspace{4 mm}
Significantly mutated genes from the cohort data are identified using the OncodriveFM
module of IntOGen. This tool detects genes that have accumulated mutations with a
high functional impact. It uses annotations from the Ensembl variant effect predictor
(VEP, V.70) that includes SIFT and Polyphen2 and precomputed MutationAssessor functional
impacts. It calculates a P value per gene from the number of mutations detected across
all possible coding bases of a gene with a positive weighting for mutations with a high
functional impact.
\end{information}

\begin{steps}
Sort the data in this file by two levels starting with the 5th column,
the FM\_QVALUE from smallest-to-biggest and then by the 12th column the
CLUST\_QVALUE also from smallest-to-biggest.
\end{steps}

The top nine genes with the smallest Q values should be
TP53, PTEN, PPP6C, CDKN2A, BRAF, NRAS, ARID2, TTN, IDH1.

\begin{information}
All of these have very small FM\_Qvalues which means they are all significantly
mutated genes in this TCGA Melanoma cohort of 338 patients.
\end{information}

\begin{steps}
Now look at their sample frequency (column 9 SAMPLE\_FREQ) values these are
the number of samples that contain at least one mutation in the gene.
\end{steps}

\begin{questions}
\begin{itemize}
\item a) Which gene has mutations in the most samples?
\item b) Which gene had the lowest FM Q value?
\item c) Why don't the genes with the lowest Q values also have the highest sample frequency
value?
\end{itemize}
\end{questions}

\begin{answer}
\begin{itemize}
\item a) TTN has the highest number of samples with mutations. There are 265 out of 327
samples with mutations in TTN.
\item b) TP53 or PTEN
\item c) The P value calculation takes into account the length of the coding sequence
of the gene, the mutation rate of the nucleotides contained within it and the functional
consequences of those changes. Therefore a small gene with a small number of deleterious
mutations may have a lower P value and also Q value than a large gene with a high
mutation frequency.
\end{itemize}
\end{answer}

\begin{information}
The results for the assessment of clustered mutations in genes carried out by the
OncodriveCLUST module of IntOGen are shown in the 10-14th columns. The Q value is in
column 12, CLUST\_QVALUE which indicates if there is a significant grouping of mutations
identified. The positions of the mutations are now described in terms of the amino acid
residue in the encoded protein.
\end{information}

The three oncogenes BRAF, NRAS and IDH1 have very low CLUST\_QVALUEs <0.01 indicating
that the mutations in these genes are highly clustered. The CLUST\_COORDS column reports
that there are 158 samples with mutations between the amino acid positions 594-601 of
BRAF; 84 samples with mutations at amino acid position 61 of NRAS; and 15 sample with
mutations at amino acid position 132 of IDH1.

\begin{questions} 
Why are the oncogenes more likely to have clustered mutations and the tumour suppressor genes less likely?
\end{questions}

\begin{answer}
Gain of function mutations are required to activate oncogenes and so only key residues in the protein
will result in activation. Tumour suppressors are frequently affected by loss of function mutations
and deletions. A truncating mutation or frameshift indel can occur in any exon, except the last one,
and have the same deleterious functional result.
\end{answer}

\begin{information}
The INTOGEN\_DRIVER column indicates if this gene is a known cancer driver gene with 1 for yes and
0 for no so it is promising to see the majority of our top hit genes are known drivers.

The XREFS column indicates a mutation match in external databases. This means the position and
the base change has been seen before. For the known Driver genes there are many COSMIC IDs indicating
these mutations have been recorded in the Catalogue of Somatic Mutations in Cancer database
\url{http://cancer.sanger.ac.uk/cosmic}. Remember the majority of data in these databases have come from
large scale cancer sequencing projects carried out by TCGA and ICGC associated groups.
\end{information}

\begin{steps}
Have a look at the values in the XREFS field for the gene TTN.
\end{steps}

There are a large number of COSMIC entries for mutations in this gene but also a large number
of ESP ID numbers that refer to the National Heart, Lung, and Blood Institute (NHLBI),
Exome Sequencing Project (ESP) \url{http://evs.gs.washington.edu/EVS/}.

\begin{information}
This data comes from a diverse population that typically don't have cancer but focus on patients
with heart, lung and blood disorders.
\end{information}

\begin{questions}
Can we draw any conclusions from the high number of ESP mutations and dbSNP references for TTN?
\end{questions}

\begin{answer}
Functional studies would be the only way to prove conclusively if TTN mutations were cancer driver mutations.
The high number of samples in the ESP cohort that have mutations in common with the cancer cohorts
could indicate that TTN mutation may not be unique to the cancer setting. This conclusion seems to be
backed up by the high number of mutations also with dbSNP ids that indicates the potential for some of
these mutations to be present in the general population.
\vspace{4 mm}
Amongst this high number of mutations there may be groups of patients where the mutations are purely
passenger and other groups where the mutations could contribute to the tumour.
\vspace{4 mm}
From this data there is no way to tell.
\end{answer}

\begin{information}
The other files in the output support the information in this sheet.
\vspace{4 mm}
The \texttt{variant\_genes.tsv} file includes a summary of all mutation found in each of the genes with a count
of the number of samples identified to have that mutation. It also reports the variant impact score
and category of which there are four; high, medium, low and none.
\end{information}

\begin{steps}
Open up the \texttt{variant\_genes.tsv} file and explore the data. 
\end{steps}

\begin{questions} 
Can you find out what the nucleotide change details for the most common BRAF mutation that results
in V600E amino acid change in the cohort?
Sort the data on the gene symbol column to make this easier.
\end{questions}

\begin{answer}
It is an A>T at position chr7:140453136 identified in 127 samples.
\end{answer}

\begin{information}
The \texttt{variant\_samples.tsv} worksheet allows you to find out the sample identification numbers
for the samples with the mutation that you are interested in.
\end{information}

\begin{steps}
Open up the \texttt{variant\_samples.tsv} file and explore the data. 
\end{steps}

\begin{questions}
Can you use the position that you found in the question above to locate the sample IDs for the 3
cases with a V600R BRAF mutation that is caused by a nucleotide change of AC/CT starting at chr7:140453136?
\end{questions}

\begin{answer}
TCGA-D3-A1QA-06A-11D-A196-08, TCGA-EB-A3XC-01A-11D-A23B-08, TCGA-ER-A193-06A-12D-A197-08
\end{answer}


%----------------------------------------------------------------------------------------
%	REFERENCES
%----------------------------------------------------------------------------------------

\section{References}

\begin{description}[style=multiline,labelindent=0cm,align=left,leftmargin=1cm]
 \item[Gunes et al. Nat. Methods 2010] \hfill\\
  \url{http://www.nature.com/nmeth/journal/v7/n2/pdf/nmeth0210-92.pdf}
 \item[Gonzalez-Perez et al. Nat. Methods 2013] \hfill\\
  \url{http://www.nature.com/nmeth/journal/v10/n11/pdf/nmeth.2642.pdf}
\end{description}
